src/clusterfuzz/_internal/bot/fuzzers/syzkaller/runner.py

```diff
@@ -145,17 +170,41 @@ def repro(self, repro_timeout, repro_args):
     return engine.ReproduceResult(result.command, result.return_code,
                                   result.time_executed, result.output)
 
-  def fuzz(self,
-           fuzz_timeout,
-           additional_args,
-           unused_additional_args=None,
-           unused_extra_env=None):
+  def save_rawcover_output(self, filepath: str):
+    """Find syzkaller port and write rawcover data to a file."""
+    port = new_process.find_localhost_port(get_syzkaller_pid())
```

Comment:
Is there a better way to get this? Like parsing some output, or passing it as an option to syzkaller? 

This is relying on a fragile assumption.

Reply:
best way to get `rawcover` data from already running `syzkaller` process, we were advised by Dmitry, is to get it from `localhost:port/rawcover`

Reply:
I mean getting the port number in a more reliable way, or pre-specifying it somehow rather than trying to figure it out this way at runtime.

Reply:
I'm not sure if there is a good way to get a port number. I can reliably get pid from your suggestion and port number should be reliably retrievable from `psutil.Popen(pid).connections()`.

There are a couple of reasons why I am getting port this way opposed to assigning syzkaller a port via config (there is a way to do this):
1. multiple instances of clusterfuzz will be running on one ATP instance. cannot set a constant port
2. pre-assignment of port can lead to race condition: one clusterfuzz spins up a syzkaller with a pre-assigned port. before syzkaller bot is instantiated, another instance of clusterfuzz spins up another syzkaller taking the port.
3. `psutil.Popen(pid)` will reliably give us ports that are only associated with the instance of syzkaller.

Reply:
`psutil.Process(pid).connections()` was emitting `AccessDeniedError`. Changed to `psutil.net_connections()` and finding a connection with a matching `pid` and localhost ip with `LISTEN`

Reply:
Ok, that sounds reasonable. Does syzkaller output anything in its logs to indicate what port it's using.

Reply:
unfortunately `syzkaller` does not log out the port it is using. It just prints what is in `config` which is unfortunately `localhost:0`. Port 0 means auto-select an available port.